### PR TITLE
Variable was unbound

### PR DESCRIPTION
**Pull request by chiarazampolli**
@davidrohr , @martenole , please check.
Related to https://github.com/AliceO2Group/O2DPG/pull/1021
Pinging also @catalinristea, this is needed for MC anchoring.

### CODE REVIEW

**Comment by chiarazampolli** [comment]
Hello @martenole , 
I am not sure I understand the CI failure. It seems related to:
```
Found 16 physical CPU cores
Launching task: o2-sim --seed 12345 -j 16 -n 100 -m PIPE ITS MFT FT0 FV0 FDD -g extgen -e TGeant4 --configKeyValues "GeneratorExternal.fileName=$O2_ROOT/share/Generators/external/QEDLoader.C;QEDGenParam.yMin=-7;QEDGenParam.yMax=7;QEDGenParam.ptMin=0.001;QEDGenParam.ptMax=1.;Diamond.width[2]=6." --run 310000 &> qedsim.log &
Launching task: o2-sim --seed 12345 -n 5 --configKeyValues Diamond.width[2]=6. -g pythia8hi -e TGeant4 -j 16 --run 310000 &> sim.log &
Launching task: o2-sim-digitizer-workflow -n 5 --simPrefixQED qed/o2sim --qed-x-section-ratio 4417.27 --disable-mc --tpc-lanes 16 --shm-segment-size 8000000000 -b --skipDet TRD --configKeyValues "HBFUtils.nHBFPerTF=128;HBFUtils.orbitFirst=0;HBFUtils.orbitFirstSampled=0;HBFUtils.obligatorySOR=false;HBFUtils.runNumber=310000;HBFUtils.maxNOrbits=128;" --interactionRate 50000 &> digi.log &
Launching task: o2-sim-digitizer-workflow -n 5 --disable-mc --onlyDet TRD --trd-digit-downscaling 1000 --shm-segment-size 8000000000 -b --incontext collisioncontext.root --configKeyValues "HBFUtils.nHBFPerTF=128;HBFUtils.orbitFirst=0;HBFUtils.orbitFirstSampled=0;HBFUtils.obligatorySOR=false;HBFUtils.runNumber=310000;HBFUtils.maxNOrbits=128;;TRDSimParams.digithreads=16;" &> digiTRD.log &
Launching task: o2-its-digi2raw --file-for cruendpoint -o raw/ITS &> itsraw.log &
Launching task: o2-mft-digi2raw --file-for cruendpoint -o raw/MFT &> mftraw.log &
Launching task: o2-ft0-digi2raw --file-for cruendpoint -o raw/FT0 &> ft0raw.log &
Launching task: o2-fv0-digi2raw --file-for cruendpoint -o raw/FV0 &> fv0raw.log &
Launching task: o2-fdd-digit2raw --file-for cruendpoint -o raw/FDD &> fddraw.log &
Launching task: o2-tpc-digits-to-rawzs --zs-version 4 --file-for cruendpoint -i tpcdigits.root -o raw/TPC &> tpcraw.log &
Launching task: o2-tof-reco-workflow -b --file-for cruendpoint --output-type raw --tof-raw-outdir raw/TOF &> tofraw.log &
Launching task: o2-mid-digits-to-raw-workflow -b --mid-raw-outdir raw/MID --file-for cruendpoint &> midraw.log &
Launching task: o2-mch-digits-to-raw --input-file mchdigits.root --output-dir raw/MCH --file-for cruendpoint &> mchraw.log &
Launching task: o2-emcal-rawcreator --file-for link -o raw/EMC &> emcraw.log &
Launching task: o2-phos-digi2raw --file-for link -o raw/PHS &> phsraw.log &
Launching task: o2-cpv-digi2raw --file-for cruendpoint -o raw/CPV &> cpvraw.log &
Launching task: o2-zdc-digi2raw --file-for cruendpoint -o raw/ZDC &> zdcraw.log &
Launching task: o2-hmpid-digits-to-raw-workflow --file-for crorcendpoint --outdir raw/HMP &> hmpraw.log &
Launching task: o2-trd-trap2raw -o raw/TRD --file-for cruendpoint &> trdraw.log &
Launching task: o2-ctp-digi2raw -o raw/CTP --file-for cruendpoint &> ctpraw.log &
Launching task: /sw/slc8_x86-64/O2/11187-slc8_x86-64-local1/prodtests/full-system-test/dpl-workflow.sh &> reco_NOGPU.log &
task timeout reached .. killing all processes
killing child 31341
```
but I don't know what it means. It seems not related to my PR, though.

File: prodtests/full-system-test/dpl-workflow.sh [hunk 1]
```diff
@@ -407,7 +407,7 @@ has_detector_reco TRD && ! has_detector_from_global_reader TRD && add_W o2-trd-t
...
 has_detector_reco TRD && [[ ! -z "$TRD_SOURCES" ]] && add_W o2-trd-global-tracking "$DISABLE_DIGIT_ROOT_INPUT $DISABLE_ROOT_OUTPUT $DISABLE_MC $TRD_CONFIG $TRD_FILTER_CONFIG --track-sources $TRD_SOURCES --pipeline $(get_N trd-globaltracking_TPC_ITS-TPC_ TRD REST 1 TRDTRK),$(get_N trd-globaltracking_TPC_FT0_ITS-TPC_ TRD REST 1 TRDTRK)" "$TRD_CONFIG_KEY;$ITSMFT_STROBES;$ITSEXTRAERR"
 has_detector_reco TOF && [[ ! -z "$TOF_SOURCES" ]] && add_W o2-tof-matcher-workflow "$DISABLE_DIGIT_ROOT_INPUT $DISABLE_ROOT_OUTPUT $DISABLE_MC --track-sources $TOF_SOURCES --pipeline $(get_N tof-matcher TOF REST 1 TOFMATCH)" "$ITSMFT_STROBES;$ITSEXTRAERR"
-has_detectors TPC && [[ -z "$DISABLE_ROOT_OUTPUT" && -z "$SKIP_TPC_CLUSTERSTRACKS_OUTPUT" ]] && add_W o2-tpc-reco-workflow "--input-type pass-through --output-type clusters,tracks,send-clusters-per-sector $DISABLE_MC"
+has_detectors TPC && [[ -z "$DISABLE_ROOT_OUTPUT" && "$SKIP_TPC_CLUSTERSTRACKS_OUTPUT" == "0" ]] && add_W o2-tpc-reco-workflow "--input-type pass-through --output-type clusters,tracks,send-clusters-per-sector $DISABLE_MC"
```
Comment (martenole) [inline prodtests/full-system-test/dpl-workflow.sh]:
```suggestion
has_detectors TPC && [[ -z "$DISABLE_ROOT_OUTPUT" && "${SKIP_TPC_CLUSTERSTRACKS_OUTPUT:-}" != 1 ]] && add_W o2-tpc-reco-workflow "--input-type pass-through --output-type clusters,tracks,send-clusters-per-sector $DISABLE_MC"
```

**Comment by martenole** [comment]
Yes, the error is unrelated, but with your changes the `SKIP_TPC_CLUSTERSTRACKS_OUTPUT` would still be unbound. It is set nowhere in our code, we want to be able to set it externally if needed. Therefore we need that parameter expansion `${SKIP_TPC_CLUSTERSTRACKS_OUTPUT:-}` which evaluates to an empty string in case `SKIP_TPC_CLUSTERSTRACKS_OUTPUT` is either not set or set to nothing

**Comment by chiarazampolli** [comment]
> Yes, the error is unrelated, but with your changes the `SKIP_TPC_CLUSTERSTRACKS_OUTPUT` would still be unbound. It is set nowhere in our code, we want to be able to set it externally if needed. Therefore we need that parameter expansion `${SKIP_TPC_CLUSTERSTRACKS_OUTPUT:-}` which evaluates to an empty string in case `SKIP_TPC_CLUSTERSTRACKS_OUTPUT` is either not set or set to nothing

Hello, 
The setting would be in O2DPG, https://github.com/AliceO2Group/O2DPG/pull/1021. Or am I misunderstanding?
The way you propose is an alternative that I considered, yes. But then I thought to define it in O2DPG. 
I can implement the one that you like the most. 
Note that the logic would change a bit, in whichever way we go: the original code was only checking the existence of the variable, now it will not work due the the check on unbound variables. So I assigned a value. 

Chiara

**Comment by chiarazampolli** [comment]
Suggestion implemented, I will close the PR in O2DPG.
Thanks, @martenole !

**Review by martenole — APPROVED** [review]

**Comment by chiarazampolli** [comment]
Hello @martenole , 
Unless I am wrong, the error is not related to the PR. Can it be merged? Both MC and async reco fail.
Thanks,
Chiara

**Pull request merged**